torch/csrc/jit/script/compiler.cpp

```diff
@@ -2408,15 +2402,35 @@ struct to_ir {
       dim++;
     };
 
+    // before ellipsis, dimension index should be `dim`
```

Comment:
the logic here is a little obscure. It's hard to track what's going on in an out of the loop, various lambdas, etc. Also, `handle_dim` is not very descriptive. Is there a way to restructure this so it's more clear for readers?

Reply:
I looked more into whether we could untangle this and there doesn't seem to be a lot we could do 😢 I added quite a few comments to explain why the slicing logic structured this particular way.

Reply:
Also renamed some variables in a hope to make it a bit clearer